### PR TITLE
Tweak self-transfer behaviour to match v15

### LINKED ISSUE #143: make sure the Machine's transfer logic is level with nv15
Concretely: https://github.com/filecoin-project/lotus/pull/7637.

See context at: https://github.com/filecoin-project/fvm/pull/137#discussion_r770105151

### PR DESCRIPTION
**Pull request by arajasek**
Closes #143

Matches behaviour in https://github.com/filecoin-project/lotus/pull/7637.

The motivation for this change is mostly around UX -- it appears weird when people successfully send themselves 2B Filecoin, even though no actual transfer occurs. This change is slightly less efficient than exiting early on self-transfers, but that's not really a case we're looking to optimize.

### CODE REVIEW

File: fvm/src/machine/default.rs [hunk 1]
```diff
@@ -214,6 +210,15 @@ where
             .context("cannot transfer from non-existent sender")
             .or_error(ErrorNumber::InsufficientFunds)?;
 
+        if from_actor.balance.lt(value) {
```
Comment (Stebalien) [inline fvm/src/machine/default.rs:213]: Can we not just use `from_actor.balance < value`?

**Review by Stebalien — APPROVED** [review]
LGTM. Do we know of any such messages on-chain? I.e., should this be behind a version guard?

NOTE: we can always remove such version guards later, but it's still useful to be able to sync/debug against chain history for now.

**Comment by jennijuju** [comment]
> LGTM. Do we know of any such messages on-chain? I.e., should this be behind a version guard?
> 
> NOTE: we can always remove such version guards later, but it's still useful to be able to sync/debug against chain history for now.

I.e https://m.filscout.com/zh/message/bafy2bzacebr2quxfina5pcy3uudh5lhrbapmbdaxd3l5qfkfhaun6yq44qxmu

**Review by raulk — CHANGES_REQUESTED** [review]
Let's add a version guard. ref-fvm officially supports nv14 and above, and this would break support for nv14. (And we should extract a test vector from that message.)

**Comment by arajasek** [comment]
Yeah, opened this before the discussion that resulted in https://github.com/filecoin-project/ref-fvm/pull/360#discussion_r821077448

**Comment by arajasek** [comment]
Pulling into #360

**Pull request closed without merging**